log4j-api/src/main/java/org/apache/logging/log4j/LogManager.java

```diff
@@ -40,7 +40,13 @@
 public class LogManager {
 
     /**
-     * Constant holding the name of the root logger (the empty string).
+     * Kept for binary compatibility.
+     */
+    @Deprecated(forRemoval = true)
+    public static final String FACTORY_PROPERTY_NAME = "log4j2.loggerContextFactory";
```

Comment:
I suppose we can keep these strings, but by 3.0.0, they'll be functionally useless due to https://github.com/apache/logging-log4j2/issues/1977

Reply:
There is actually a compatibility mechanism developed by Ralph (cf. `log4j2.propertyMapping.json`) that allows `log4j2.loggerContextFactory` to still work.

I had doubts about reintroducing this: it is a **compile-time** constant, so it will be inlined in all binary artifacts. As far as source compatibility is concerned, I think we could remove it. What do you think?

Reply:
What I mean is that specifying custom classes to use for things will likely use some sort of code-based way of doing so rather than a system property.